### PR TITLE
mcu: send one text event per line on Nano X

### PR DESCRIPTION
**Pull request by greenknot**
This is already the case on Nano S.

### CODE REVIEW

File: mcu/seproxyhal.py [hunk 1]
```diff
@@ -187,8 +187,10 @@ def _send_packet(self, tag, data=b''):
 
     def apply_automation(self, text, x, y):
         if self.automation_server:
-            event = {"text": text.decode("ascii", "ignore"), "x": x, "y": y}
-            self.automation_server.broadcast(event)
+            lines = text.decode("ascii", "ignore").split("\n")
+            for line in lines:
+                event = {"text": line, "x": x, "y": y}
```
Comment (niooss-ledger) [inline mcu/seproxyhal.py:192]: I am not familiar with this code, but shouldn't `y` be updated for each line?

Comment (greenknot) [reply mcu/seproxyhal.py]: Thanks for that feedback. Indeed, it's wrong. I pushed force a new commit which is a bit more complex but handles correctly that.

File: mcu/nanox_ocr.py [hunk 2]
```diff
@@ -151,19 +151,20 @@ def analyze_bitmap(self, data: bytes):
...
 
-        self.last_x, self.last_y = x, y
+        self.last_y = y
 
-    def get_text(self, reset=True) -> Tuple[bytes, Tuple[int, int]]:
-        text, x, y = self.text, self.x, self.y
-        if reset:
-            self.reset()
-        return text, (x, y)
+    def get_events(self) -> Tuple[bytes, Tuple[int, int]]:
```
Comment (niooss-ledger) [inline mcu/nanox_ocr.py:167]: The return type must be updated

File: mcu/nanox_ocr.py [hunk 3]
```diff
@@ -152,18 +151,15 @@ def analyze_bitmap(self, data: bytes):
...
-        if reset:
-            self.reset()
-        return text, (x, y)
+            # create a new TextEvent if there are no events yet or if there is a new line
+            if not self.events or y > self.events[-1].y:
+                event = TextEvent("", x, y)
+                self.events.append(event)
+            else:
+                event = self.events[-1]
+            event.text += char
```
Comment (niooss-ledger) [inline mcu/nanox_ocr.py:160]: Or even simpler:
```python
            if not self.events or y > self.events[-1].y:
                self.events.append(TextEvent(char, x, y))
            else:
                self.events[-1].char += char
```

... which becomes even more readable by inversing the condition:

```python
            if self.events and y <= self.events[-1].y:
                self.events[-1].char += char
            else:
                self.events.append(TextEvent(char, x, y))
```

**Review by niooss-ledger — APPROVED** [review]

**Pull request merged**